app/src/main/java/com/duckduckgo/app/location/data/LocationPermissionEntity.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.location.data
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Entity(tableName = "locationPermissions")
+data class LocationPermissionEntity(
+    @PrimaryKey val domain: String,
+    val permission: LocationPermissionType
+)
+
+private const val HTTPS_HOST_PREFIX = "https://"
+private const val WWW_SUFFIX = "/"
+
+fun LocationPermissionEntity.forFireproofing(): String {
```

Comment:
This seems almost identical to the function in `StringHtmlExtension`. Worth combining?

Reply:
not the same though, `LocationPermissionEntity` domains don't have `www` (because the fireproofed websites are stored like that) but the ones coming from the `GeolocationAPI` do. So I wanted to keep them separated